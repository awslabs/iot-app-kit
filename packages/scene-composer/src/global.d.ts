declare module '*.hdr';
declare module '*.png';
