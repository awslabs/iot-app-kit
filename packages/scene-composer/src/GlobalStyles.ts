import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  .sidePanelTabs {
    overflow-y: auto;
  }
  .sidePanelTabs > div:first-of-type {
    padding: 0 !important; // This removes the padding of the Polaris Tabs Component
  }`;
