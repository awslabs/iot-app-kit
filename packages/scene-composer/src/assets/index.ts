export * from './anchors';
export * from './viewpoints';
