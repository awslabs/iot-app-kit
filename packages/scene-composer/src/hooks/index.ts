export * from './useTween';
export * from './useEditorHelper';
