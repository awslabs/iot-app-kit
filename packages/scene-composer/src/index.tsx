export * from './interfaces';

export * from './errors';

export * from './SceneComposer';

// Augmentation related interfaces
export * from './augmentations/three';

export * from './GlobalSettings';

export * from './utils/dataFrameLabelUtils';

export * from './utils/dataBindingTemplateUtils';

export * from './constants';
