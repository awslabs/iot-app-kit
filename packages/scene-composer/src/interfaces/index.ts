export * from './assets';
export * from './interfaces';
export * from './components';
export * from './dataTypes';
export * from './dataBinding';
export * from './metricRecorder';
export * from './feature';
export * from './sceneViewer';
export * from './componentSettings';
