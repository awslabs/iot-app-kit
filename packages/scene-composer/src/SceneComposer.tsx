import React, { useMemo } from 'react';
import { ThemeProvider } from 'styled-components';

import LogProvider from './logger/react-logger/log-provider';
import IntlProvider from './components/IntlProvider';
import { darkTheme, lightTheme } from './theme';
import { GlobalStyles } from './GlobalStyles';
import { useStore } from './store';
import { sceneComposerIdContext } from './sceneComposerIdContext';
import { generateUUID } from './utils/mathUtils';
import StateManager, { SceneComposerProps } from './StateManager';
import DefaultErrorFallback from './DefaultErrorFallback';
import { SCENE_BODY_CLASS } from './constants';

export const SceneComposer: React.FC<SceneComposerProps> = ({
  sceneComposerId,
  ErrorView,
  onError,
  config,
  locale,
  ...props
}: SceneComposerProps) => {
  const currentSceneComposerId = useMemo(() => sceneComposerId ?? generateUUID(), [sceneComposerId]);

  const ErrorFallback: React.ReactNode = ErrorView || DefaultErrorFallback;

  // Assuming we only have dark or light...
  const theme = config.colorTheme ?? config.colorTheme === 'light' ? lightTheme : darkTheme;

  // label body as being used by the scene. this allows other components to identify it vs. other page components
  document.body.className = document.body.className + ' ' + SCENE_BODY_CLASS;

  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      <LogProvider namespace='SceneComposer' logger={config.logger} ErrorView={ErrorFallback} onError={onError}>
        <IntlProvider locale={locale}>
          <sceneComposerIdContext.Provider value={currentSceneComposerId}>
            <StateManager config={config} locale={locale} {...props} />
          </sceneComposerIdContext.Provider>
        </IntlProvider>
      </LogProvider>
    </ThemeProvider>
  );
};

export function useSceneComposerApi(sceneComposerId: string) {
  const state = useStore(sceneComposerId).getState();
  return {
    findSceneNodeRefBy: state.findSceneNodeRefBy,
    setCameraTarget: state.setCameraTarget,
    setSelectedSceneNodeRef: state.setSelectedSceneNodeRef,
  };
}

export type SceneComposerApi = ReturnType<typeof useSceneComposerApi>;
