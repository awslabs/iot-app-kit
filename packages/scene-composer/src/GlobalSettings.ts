import { COMPOSER_FEATURES, FeatureConfig, GetSceneObjectFunction } from './interfaces';
import { IMetricRecorder } from './interfaces/metricRecorder';

export interface DracoDecoderConfig {
  enable: boolean;
  path?: string;
}

const globalSettings: {
  debugMode: boolean;
  dracoDecoder: DracoDecoderConfig;
  locale: string;
  cdnPath: string | undefined;
  shouldEnableDataBindingTemplate: boolean;
  metricRecorder?: IMetricRecorder;
  featureConfig: FeatureConfig;
  getSceneObjectFunction: GetSceneObjectFunction | undefined;
} = {
  debugMode: false,
  dracoDecoder: { enable: true },
  locale: 'en-US',
  cdnPath: undefined,
  shouldEnableDataBindingTemplate: false,
  metricRecorder: undefined,
  // default to disable all features
  featureConfig: {},
  getSceneObjectFunction: undefined,
};

const changeSubscribers = [] as Function[];

const notifySubscribers = () => {
  changeSubscribers.forEach((sub) => sub());
};

export const setDebugMode = () => {
  globalSettings.debugMode = true;
  notifySubscribers();
};

export const setDracoDecoder = (dracoDecoder: DracoDecoderConfig) => {
  globalSettings.dracoDecoder = dracoDecoder;
  notifySubscribers();
};
export const setLocale = (locale: string) => {
  globalSettings.locale = locale;
  notifySubscribers();
};

export const setCdnPath = (path: string | undefined) => {
  globalSettings.cdnPath = path;
  notifySubscribers();
};

// Before we have feature flag, we use global settings to control this new feature
export const enableDataBindingTemplate = () => {
  globalSettings.shouldEnableDataBindingTemplate = true;
  notifySubscribers();
};

export const setMetricRecorder = (metricRecorder: IMetricRecorder) => {
  globalSettings.metricRecorder = metricRecorder;
  notifySubscribers();
};

export const setFeatureConfig = (featureConfig: Partial<Record<COMPOSER_FEATURES, boolean>>) => {
  globalSettings.featureConfig = featureConfig;
  notifySubscribers();
};

export const setGetSceneObjectFunction = (getSceneObjectFunction: GetSceneObjectFunction) => {
  globalSettings.getSceneObjectFunction = getSceneObjectFunction;
  notifySubscribers();
};

export const getGlobalSettings = (): Readonly<typeof globalSettings> => {
  return globalSettings;
};

export const subscribe = (callback: Function) => {
  changeSubscribers.push(callback);
};

export const unsubscribe = (callback: Function) => {
  changeSubscribers.splice(changeSubscribers.indexOf(callback), 1);
};
