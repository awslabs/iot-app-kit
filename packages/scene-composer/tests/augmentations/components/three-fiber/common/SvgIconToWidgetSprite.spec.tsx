import renderer from 'react-test-renderer';
import React from 'react';

import { DefaultAnchorStatus, SelectedAnchor } from '../../../../../src';
import {
  ErrorIconSvgString,
  InfoIconSvgString,
  SelectedIconSvgString,
  VideoIconSvgString,
  WarningIconSvgString,
} from '../../../../../src/assets';
import svgIconToWidgetSprite from '../../../../../src/augmentations/components/three-fiber/common/SvgIconToWidgetSprite';

describe('svgIconToWidgetSprite', () => {
  [
    ['Selected', { key: SelectedAnchor, icon: SelectedIconSvgString }, false],
    ['Info', { key: DefaultAnchorStatus.Info, icon: InfoIconSvgString }, false],
    ['Warning', { key: DefaultAnchorStatus.Warning, icon: WarningIconSvgString }, false],
    ['Error', { key: DefaultAnchorStatus.Error, icon: ErrorIconSvgString }, false],
    ['Video', { key: DefaultAnchorStatus.Video, icon: VideoIconSvgString }, false],
    ['Selected', { key: SelectedAnchor, icon: SelectedIconSvgString }, true],
    ['Info', { key: DefaultAnchorStatus.Info, icon: InfoIconSvgString }, true],
    ['Warning', { key: DefaultAnchorStatus.Warning, icon: WarningIconSvgString }, true],
    ['Error', { key: DefaultAnchorStatus.Error, icon: ErrorIconSvgString }, true],
    ['Video', { key: DefaultAnchorStatus.Video, icon: VideoIconSvgString }, true],
  ].forEach((value) => {
    it(`it should render the ${value[2] ? 'always visible' : ''} ${value[0]} correctly`, () => {
      jest.spyOn(window.Math, 'random').mockReturnValue(0.1);
      const { key, icon } = value[1] as any;
      const container = renderer.create(svgIconToWidgetSprite(icon, key, value[2] as boolean));

      expect(container).toMatchSnapshot();
    });
  });
});
