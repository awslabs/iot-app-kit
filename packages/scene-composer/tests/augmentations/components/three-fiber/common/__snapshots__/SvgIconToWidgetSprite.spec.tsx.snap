// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svgIconToWidgetSprite it should render the  Error correctly 1`] = `
<widgetSprite
  name="Error"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the  Info correctly 1`] = `
<widgetSprite
  name="Info"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the  Selected correctly 1`] = `
<widgetSprite
  name="Selected"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the  Video correctly 1`] = `
<widgetSprite
  name="Video"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the  Warning correctly 1`] = `
<widgetSprite
  name="Warning"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Error correctly 1`] = `
<widgetSprite
  name="Error"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Info correctly 1`] = `
<widgetSprite
  name="Info"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Selected correctly 1`] = `
<widgetSprite
  name="Selected"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Video correctly 1`] = `
<widgetSprite
  name="Video"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;

exports[`svgIconToWidgetSprite it should render the always visible Warning correctly 1`] = `
<widgetSprite
  name="Warning"
>
  <primitive
    attach="visual"
    object={
      Object {
        "materials": Array [
          Object {
            "color": 16777215,
            "colorWrite": true,
            "depthFunc": 3,
            "depthTest": true,
            "depthWrite": true,
            "sizeAttenuation": true,
            "stencilFail": 7680,
            "stencilFunc": 519,
            "stencilFuncMask": 255,
            "stencilRef": 0,
            "stencilWrite": false,
            "stencilWriteMask": 255,
            "stencilZFail": 7680,
            "stencilZPass": 7680,
            "transparent": true,
            "type": "SpriteMaterial",
            "uuid": "99999919-9999-4919-9999-991999999919",
            "wireframeLinecap": undefined,
            "wireframeLinejoin": undefined,
          },
        ],
        "metadata": Object {
          "generator": "Object3D.toJSON",
          "type": "Object",
          "version": 4.5,
        },
        "object": Object {
          "children": Array [
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
            Object {
              "layers": 1,
              "material": "99999919-9999-4919-9999-991999999919",
              "matrix": Array [
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
                0,
                0,
                0,
                0,
                1,
              ],
              "renderOrder": 10,
              "type": "Sprite",
              "uuid": "99999919-9999-4919-9999-991999999919",
            },
          ],
          "layers": 1,
          "matrix": Array [
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
            0,
            0,
            0,
            0,
            1,
          ],
          "type": "Group",
          "uuid": "99999919-9999-4919-9999-991999999919",
        },
      }
    }
  />
</widgetSprite>
`;
